docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -0,0 +1,403 @@
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: filebeat-quickstart
+spec:
+  config:
+    filebeat:
+      autodiscover:
+        providers:
+        - host: ${HOSTNAME}
+          type: kubernetes
+          hints:
+            enabled: true
+            default_config:
+              type: container
+              paths:
+              - /var/log/containers/*${data.kubernetes.container.id}.log
+  daemonSet:
+    podTemplate:
+      spec:
+        serviceAccount: elastic-beat-filebeat-quickstart
+        automountServiceAccountToken: true
+...
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: elastic-beat-filebeat-quickstart
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: elastic-beat-autodiscover-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: elastic-beat-autodiscover
+subjects:
+- kind: ServiceAccount
+  name: elastic-beat-filebeat-quickstart
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: elastic-beat-autodiscover
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  - namespaces
+  - events
+  - pods
+  verbs:
+  - get
+  - list
+  - watch
+----
+
+[id="{p}-beat-deploying-beats-in-secured-clusters"]
+=== Deploying Beats in secured clusters
+
+To deploy Beats in clusters with Pod Security Policy admission controller enabled, or in OpenShift clusters, you have to grant additional permissions to the Service Account that Beat Pods are running with. Those Service Accounts need to be bound to a Role or ClusterRole that has `use` permission for the required PSP/SCC. Different Beats and their features might require different settings set in their PSP/SCC.
+
+[id="{p}-beat-troubleshooting"]
+== Troubleshooting
+
+[id="{p}-beat-beat-pods-are-crashing-when-kibanaref-is-specified"]
+=== Beat Pods are crashing when kibanaRef is specified
+
+When `kibanaRef` is specified, Beat tries to connect to the Kibana instance. If it's unable to do so, the Pod crashes. This may happen when Kibana is not yet up or when Beat user is not yet created in Elasticsearch. The pod may cycle at most few times during the first initialization. Afterwards, the Beat should run successfully.
+
+
+[id="{p}-beat-configuration-containing-key-null-is-malformed"]
+=== Configuration containing key: null is malformed
+
+When Beat config containing `some_key: null` is applied using `kubectl apply` it's interpreted by `kubectl` diff mechanism as an instruction to remove key `some_key` from the config. This is unfortunate, but works according to the spec it follows. To overcome that, `null` can sometimes be replaced by an empty dictionary, like: `some_key: {}`. If this is not possible due to a particular settings semantics, `configRef` field can be used instead of `config` as Secret data doesn't go through that diff process.
```

Comment:
@alaudazzi How about this?

```suggestion
When `kubectl` is used to modify a resource, it calculates a diff between the user applied and the already existing config. This diff has a special semantics link:https://tools.ietf.org/html/rfc7396#section-1[defined], that allow forcing removal of a particular key by setting its value to `null`. When user applied config contains `some_key: null` (or equivalent `some_key: ~`) it's interpreted as an instruction to remove `some_key`, often resulting in a malformed config. In Beat configs, this is often the case with link:https://www.elastic.co/guide/en/beats/filebeat/current/add-cloud-metadata.html[processors]. To avoid this:

- change `some_key: null` to `some_key: {}` - if it doesn't change config semantics
- instead of `config` use `configRef` and reference a Secret containing the config - Secret data doesn't go through the same diffing process
```

Reply:
With more details, it's definitely better.
For 
`that allow forcing removal of a particular key by setting its value to `null``
maybe we can reword into
`that allows you to force the removal of a particular key by setting its value to `null`
?